test/string_view-test/string_view-test.cpp

```diff
@@ -0,0 +1,349 @@
+/*
+* string_view-test.cpp
+* Ray Chen, Sean Murthy 
+* (c) 2020 sigcpp https://sigcpp.github.io. See LICENSE.MD
+*
+* Attribution and copyright notice must be retained.
+* - Attribution may be augmented to include additional authors
+* - Copyright notice cannot be altered
+* Attribution and copyright info may be relocated but they must be conspicuous.
+*
+* Test the string_view of the C++ standard library and 
+* provide a test benchmark for subsequent implementation of string_view.
+* - see C++17 [string.view]
+*	https://timsong-cpp.github.io/cppwp/n4659/string.view
+*/
+
+#include <stdexcept>
+
+#include "../verifiers.h"
+
+void string_view_test()
+{
+	using std::string_view;
+
+	//1. Construction and assignment
+	//2. Capacity
+
+	// Default constructor. Constructs an empty string_view. 
```

Comment:
I strongly recommend including test for `data` member in each of the following test cases: default ctor, copy ctor, the two custom ctors, and assignment.

Reply:
I'm still a little confused. Shouldn't we test the `data` after the `non-member comparison functions` like `==` is correct? Because we need `==` to judge whether the value of data is as expected, such as `sv.data() == somevalue` or `sv_2[0] == somechar`.

Reply:
Perhaps I am mistaken, but consider this code. This code is testing if `data()` does the right thing. I don't understand the dependency you say exists with some test later.

PS: I'm away and expect to return by 9pm

```cpp
const char z[]{"he"};
std::string_view sv{z};
bool b = sv.data() == z;
```

Reply:
yes, in your code. the `sv.data() == z` return a bool value. but how do we know the return value is correct? so, in this case, it dependency the operator `==`, the return value is correct only when operator `==` is overloaded correctly. So I have to test `non-member comparison functions` first.

Reply:
I feel there is a misunderstanding. The `==` operator here is **not** sv's overloaded. Instead, it is the built in `==` operator and is just comparing two pointers. The thing about whether the content at the location `data()` returns is tested later with iterators, subscript, and `at`. That is the only way we can test if the data content is correct. Except, here we only want to know if the pointer we gave sv at creation is the pointer that `data()` returns.